### PR TITLE
vulnfeeds: Get list of upstream versions and correlate extracted versions.

### PR DESCRIPTION
**Pull request by oliverchang**
Extract a list of upstream versions and use this to correlate with the
versions we extract from the CVE to ensure accuracy.

Currently just prints if we get a mismatch.

### CODE REVIEW

**Comment by inferno-chromium** [comment]
Adding @asraa team member for review mix, please feel free to assign back if code is confusing. This code is for generating python vulnerability database from cve data feed.

File: vulnfeeds/cves/versions.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func extractGitHubCommit(link string) *FixCommit {
 	}
 }
 
-func ExtractVersionInfo(cve CVEItem) VersionInfo {
+func ExtractVersionInfo(cve CVEItem, validVersions []string) VersionInfo {
```
Comment (asraa) [inline vulnfeeds/cves/versions.go:65]: is validVersions meat to be used later? (if so, TODO?)

File: vulnfeeds/pypi/pypi.go [hunk 2]
```diff
@@ -89,14 +118,34 @@ func NewPyPI(pypiJSON string) *PyPI {
...
+// processVersions takes a pypi_versions.json and returns a map of packages to versions.
+func processVersions(versionsSource []pypiVersions) map[string][]string {
+	versions := map[string][]string{}
+	for _, data := range versionsSource {
+		versions[data.Name] = data.Versions
+	}
+	return versions
+}
+
+func New(pypiLinksJSON string, pypiVersionsJSON string) *PyPI {
```
Comment (asraa) [inline vulnfeeds/pypi/pypi.go:133]: love the cleanup!

**Review by asraa — COMMENTED** [review]
I can't comment too much about the Go code, but thanks!

File: vulnfeeds/cves/versions.go (hunk 1 above)
Comment (oliverchang) [reply vulnfeeds/cves/versions.go:65]: Yep! I had some TODOs already in this function but I'll clarify them to mention this variable explicitly.

**Review by asraa — APPROVED** [review]
LGTM!
(adding @inferno-chromium, because I am unsure of my Go style reviewability)

**Pull request merged**